src/main/java/de/mediathekview/mserver/crawler/basic/M3U8Dto.java

```diff
@@ -0,0 +1,86 @@
+package de.mediathekview.mserver.crawler.basic;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Datenstruktur für M3U8-Datei
+ */
+public class M3U8Dto {
+
+  private final String url;
+  private final Map<String, String> meta = new HashMap<>();
+
+  public M3U8Dto(String aUrl) {
+    url = aUrl;
+  }
+
+  public String getUrl() {
+    return url;
+  }
+
+  public void addMeta(String key, String value) {
+    meta.put(key, value);
+  }
+
+  public Optional<String> getMeta(String key) {
+    if (!meta.containsKey(key)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(meta.get(key));
+  }
+
+  @Override
+  public boolean equals(final Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (getClass() != obj.getClass()) {
+      return false;
+    }
+    final M3U8Dto other = (M3U8Dto) obj;
+    if (url == null) {
+      if (other.url != null) {
+        return false;
+      }
+    } else if (!url.equals(other.url)) {
+      return false;
+    }
+
+    if (meta.size() != other.meta.size()) {
+      return false;
+    }
+    
+    for (String key : meta.keySet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)